runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -402,6 +405,14 @@ private void setTimer(WindowedValue<InputT> timerElement, TimerInternals.TimerDa
 
   @Override
   public void fireTimer(InternalTimer<?, TimerInternals.TimerData> timer) {
+    if (CleanupTimer.GC_TIMER_ID.equals(timer.getNamespace().getTimerId())) {
```

Comment:
It would be great if we could just accumulate the state ids that need to be cleaned up after the bundle finishes, instead of an indirection for timer execution.

Reply:
I removed the extra method but it needs to be executed via `fireTimer` to fit with `CleanupTimer`. This could be improved separately, outside of this PR.

Reply:
See my comment above. I'd revert to the old version. I'm ok with refactoring this in a follow-up.

Reply:
Reverted. Currently the path through `CleanupTimer` runner is needed to setup the context. That would need to be refactored.